<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.15.0" xml:lang="en-US">
  <compounddef id="class_luthetus_1_1_common_1_1_razor_lib_1_1_exceptions_1_1_luthetus_unit_test_exception" kind="class" language="C#" prot="public">
    <compoundname>Luthetus::Common::RazorLib::Exceptions::LuthetusUnitTestException</compoundname>
    <basecompoundref refid="class_luthetus_1_1_common_1_1_razor_lib_1_1_exceptions_1_1_luthetus_exception" prot="public" virt="non-virtual">Luthetus.Common.RazorLib.Exceptions.LuthetusException</basecompoundref>
    <sectiondef kind="public-func">
      <memberdef kind="function" id="class_luthetus_1_1_common_1_1_razor_lib_1_1_exceptions_1_1_luthetus_unit_test_exception_1a40300101f4f74c92c1c19196223f7599" prot="public" static="no" const="no" explicit="no" inline="yes" virt="non-virtual">
        <type></type>
        <definition>Luthetus.Common.RazorLib.Exceptions.LuthetusUnitTestException.LuthetusUnitTestException</definition>
        <argsstring>(string? message=null, Exception? innerException=null)</argsstring>
        <name>LuthetusUnitTestException</name>
        <qualifiedname>Luthetus.Common.RazorLib.Exceptions.LuthetusUnitTestException.LuthetusUnitTestException</qualifiedname>
        <param>
          <type>string?</type>
          <declname>message</declname>
          <defval>null</defval>
        </param>
        <param>
          <type>Exception?</type>
          <declname>innerException</declname>
          <defval>null</defval>
        </param>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="Source/Lib/Common/Exceptions/LuthetusUnitTestException.cs" line="12" column="12" bodyfile="Source/Lib/Common/Exceptions/LuthetusUnitTestException.cs" bodystart="12" bodyend="16"/>
      </memberdef>
    </sectiondef>
    <briefdescription>
    </briefdescription>
    <detaileddescription>
<para>This class is an exception to the naming convention, &quot;don&apos;t use the word &apos;Luthetus&apos; in class names&quot;.</para>
<para>Reason for this exception: the &apos;Exception&apos; datatype is far more common in code, than some specific type (example: DialogDisplay.razor). So, adding &apos;<ref refid="namespace_luthetus" kindref="compound">Luthetus</ref>&apos; in the class name for redundancy seems meaningful here. </para>
    </detaileddescription>
    <inheritancegraph>
      <node id="3">
        <label>Exception</label>
      </node>
      <node id="1">
        <label>Luthetus.Common.RazorLib.Exceptions.LuthetusUnitTestException</label>
        <link refid="class_luthetus_1_1_common_1_1_razor_lib_1_1_exceptions_1_1_luthetus_unit_test_exception"/>
        <childnode refid="2" relation="public-inheritance">
        </childnode>
      </node>
      <node id="2">
        <label>Luthetus.Common.RazorLib.Exceptions.LuthetusException</label>
        <link refid="class_luthetus_1_1_common_1_1_razor_lib_1_1_exceptions_1_1_luthetus_exception"/>
        <childnode refid="3" relation="public-inheritance">
        </childnode>
      </node>
    </inheritancegraph>
    <collaborationgraph>
      <node id="3">
        <label>Exception</label>
      </node>
      <node id="1">
        <label>Luthetus.Common.RazorLib.Exceptions.LuthetusUnitTestException</label>
        <link refid="class_luthetus_1_1_common_1_1_razor_lib_1_1_exceptions_1_1_luthetus_unit_test_exception"/>
        <childnode refid="2" relation="public-inheritance">
        </childnode>
      </node>
      <node id="2">
        <label>Luthetus.Common.RazorLib.Exceptions.LuthetusException</label>
        <link refid="class_luthetus_1_1_common_1_1_razor_lib_1_1_exceptions_1_1_luthetus_exception"/>
        <childnode refid="3" relation="public-inheritance">
        </childnode>
      </node>
    </collaborationgraph>
    <location file="Source/Lib/Common/Exceptions/LuthetusUnitTestException.cs" line="10" column="7" bodyfile="Source/Lib/Common/Exceptions/LuthetusUnitTestException.cs" bodystart="11" bodyend="17"/>
    <listofallmembers>
      <member refid="class_luthetus_1_1_common_1_1_razor_lib_1_1_exceptions_1_1_luthetus_exception_1aa5813c0119585ffef7ef8b55e7f1e081" prot="public" virt="non-virtual"><scope>Luthetus::Common::RazorLib::Exceptions::LuthetusUnitTestException</scope><name>LuthetusException</name></member>
      <member refid="class_luthetus_1_1_common_1_1_razor_lib_1_1_exceptions_1_1_luthetus_unit_test_exception_1a40300101f4f74c92c1c19196223f7599" prot="public" virt="non-virtual"><scope>Luthetus::Common::RazorLib::Exceptions::LuthetusUnitTestException</scope><name>LuthetusUnitTestException</name></member>
    </listofallmembers>
  </compounddef>
</doxygen>
